### ISSUE TITLE
CORS error on .well-known/openid-configuration for Org Authorization Server

### ISSUE THREAD

**Issue by sbauch** [okta/okta-auth-js#553]
Seems similar to #297, opening a new issue per the request there.

I'm wanting to create an OAuth flow against the Org Authorization Server such that I can get an access token that can be used against Okta's REST API.

>You can now interact with Okta APIs using scoped OAuth 2.0 access tokens for a number of Okta endpoints. Each access token enables the bearer to perform specific actions on specific Okta endpoints, with that ability controlled by which scopes the access token contains.
(https://developer.okta.com/docs/reference/api-overview/#authentication)

My understanding is that this set of options would allow me to authorize my Okta users against the Org Authorization Server and the resulting access token would be able to be used against the REST API (to manage apps, users, etc) -

```typescript
  const authClient = new OktaAuth({
    pkce: true,
    clientId: process.env.OKTA_CLIENT_ID,
    issuer: 'https://dev-162024.okta.com',
    redirectUri: 'http://localhost:3000/implicit/callback',
    scopes: ['openid', 'profile', 'email', 'okta.users.manage', 'okta.apps.manage'],
  });
```

But when I call `authClient.token.getWithPopup()`, I get a CORS error:
```console
Access to XMLHttpRequest at 'https://dev-162024.okta.com/.well-known/openid-configuration' from origin 'http://localhost:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

if I use the "Default Custom Authorization Server", the discovery endpoints do support CORS, and I am able to authenticate via a popup and get an access token, but that token can't be used to call the Okta REST API, which is expected.

```typescript
  const authClient = new OktaAuth({
    pkce: true,
    clientId: process.env.OKTA_CLIENT_ID,
    issuer: 'https://dev-162024.okta.com/oauth2/default',
    redirectUri: 'http://localhost:3000/implicit/callback',
    scopes: ['openid', 'profile', 'email'], // "Only the Org Authorization Server can mint access tokens that contain Okta API scopes."
  });
```

I am able to use Postman to follow the [instructions here](https://developer.okta.com/docs/guides/implement-oauth-for-okta/request-access-token/) to fetch an access token from the Org Authorization Server and make API calls.

I haven't found a workaround yet - I tried providing a the [CustomUrls](https://github.com/okta/okta-auth-js/blob/b441ee0bb832e806e41a4f7046337976ca565886/lib/types/OktaAuthOptions.ts#L34-L41) to the config that I got from the discovery endpoint to no avail.

```typescript
    authorizeUrl: 'https://dev-162024.okta.com/oauth2/v1/authorize',
    userinfoUrl: 'https://dev-162024.okta.com/oauth2/v1/userinfo',
    tokenUrl: 'https://dev-162024.okta.com/oauth2/v1/token',
    revokeUrl: 'https://dev-162024.okta.com/oauth2/v1/revoke',
    logoutUrl: 'https://dev-162024.okta.com/oauth2/v1/logout',
    issuer: 'https://dev-162024.okta.com',
```

Is this just not supported for client side authorization code grant flows? I'm a little confused by issues in this repository and some Okta docs / blog posts that discuss implicit flow vs. pkce authorization code grant flow and which is currently supported / recommended.

**Comment by aarongranick-okta** [comment]
@sbauch have you added `http://localhost:3000` to the "trusted origins"? https://dev-162024-admin.okta.com/admin/access/api/trusted_origins

if the trusted origins is setup, can you try clearing all browser storage for `localhost`? It is possible the issue may be caused by a corrupt browser cache.

**Comment by sbauch** [comment]
smh yup that does it 🙈 

thanks Aaron, very helpful. I'm still a bit confused about some aspects of the Okta APIs for the integration im hoping to build.

I'd like for any Okta user to be able to authenticate their Okta account to perform actions inside their organization - should we close this and I'll post about that on the dev forum?

**Comment by aarongranick-okta** [comment]
@sbauch What you describe should be possible. Of course the actions a user can take are limited by the scopes that user has access to. But a user should be able to modify their own properties and access other resources, if permitted. Administrator users have greater permissions. The set of built-in scopes provided by the Org authorization server are listed here: https://developer.okta.com/docs/guides/implement-oauth-for-okta/scopes/ These allow users to access and manage Okta resources.

Of course, you can also use a custom authentication server and define your own scopes, which can be used to secure your own APIs. 

I'm glad your immediate issue was resolved. Thanks again for your submission and please feel free to create a new issue if you find any fault with our SDK.
